maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -123,107 +135,177 @@ public fun GoogleMap(
+    }
+}
+
+// TODO make private
+internal fun MapView.tagData(): MapTagData = (tag as? MapTagData) ?: let { mapView ->
+    MapTagData(null, null, null).also { newTag ->
+        mapView.tag = newTag
     }
 }
 
-private suspend inline fun MapView.newComposition(
-    parent: CompositionContext,
+private suspend fun MapView.createComposition(
     mapClickListeners: MapClickListeners,
-    noinline content: @Composable () -> Unit
+    parentComposition: CompositionContext
 ): Composition {
     val map = awaitMap()
     return Composition(
-        MapApplier(map, this, mapClickListeners), parent
-    ).apply {
-        setContent(content)
-    }
+        applier = MapApplier(map, this, mapClickListeners),
+        parent = parentComposition
+    )
 }
 
-/**
- * Registers lifecycle observers to the local [MapView].
- */
-@Composable
-private fun MapLifecycle(mapView: MapView) {
-    val context = LocalContext.current
-    val lifecycle = LocalLifecycleOwner.current.lifecycle
-    val previousState = remember { mutableStateOf(Lifecycle.Event.ON_CREATE) }
-    DisposableEffect(context, lifecycle, mapView) {
-        val mapLifecycleObserver = mapView.lifecycleObserver(previousState)
-        val callbacks = mapView.componentCallbacks()
-
-        lifecycle.addObserver(mapLifecycleObserver)
-        context.registerComponentCallbacks(callbacks)
-
-        onDispose {
-            lifecycle.removeObserver(mapLifecycleObserver)
-            context.unregisterComponentCallbacks(callbacks)
-        }
-    }
-    DisposableEffect(mapView) {
-        onDispose {
-            mapView.onDestroy()
-            mapView.removeAllViews()
-        }
-    }
+internal fun MapView.log(msg: String, tag: String = TAG) {
+    Log.d(tag, "[MapView/${tagData().debugId}] $msg")
 }
 
-private fun MapView.lifecycleObserver(previousState: MutableState<Lifecycle.Event>): LifecycleEventObserver =
-    LifecycleEventObserver { _, event ->
-        event.targetState
-        when (event) {
-            Lifecycle.Event.ON_CREATE -> {
-                // Skip calling mapView.onCreate if the lifecycle did not go through onDestroy - in
-                // this case the GoogleMap composable also doesn't leave the composition. So,
-                // recreating the map does not restore state properly which must be avoided.
-                if (previousState.value != Lifecycle.Event.ON_STOP) {
-                    this.onCreate(Bundle())
-                }
-            }
-
-            Lifecycle.Event.ON_START -> this.onStart()
-            Lifecycle.Event.ON_RESUME -> this.onResume()
-            Lifecycle.Event.ON_PAUSE -> this.onPause()
-            Lifecycle.Event.ON_STOP -> this.onStop()
-            Lifecycle.Event.ON_DESTROY -> {
-                //handled in onDispose
-            }
-
-            else -> throw IllegalStateException()
-        }
-        previousState.value = event
-    }
-
 private fun MapView.componentCallbacks(): ComponentCallbacks =
```

Comment:
This does not need to be a separate function, merge into registerAndSaveComponentCallbacks.